pandas/io/orc.py

```diff
@@ -209,6 +210,21 @@ def to_orc(
     if engine_kwargs is None:
         engine_kwargs = {}
 
+    # validate index
+    # --------------
+
+    # validate that we have only a default index
+    # raise on anything else as we don't serialize the index
+
+    if not df.index.equals(RangeIndex.from_range(range(len(df)))):
```

Comment:
Might be clearer to use `pandas.core.indexes.api.default_index`

Reply:
Updated